test/api.jl

```diff
@@ -115,5 +116,30 @@ recommendation = provide_recommendation(trial_vector)
 recommendation_multi = provide_recommendation(trial_vector_multihyperparams)
 @test recommendation_multi == Dict(:x => 3, :y => 7)
 
+# Test case reproducing the issue in “obs_memo” function from the higher level
+space = Dict(
+    :a => HP.Choice(:a, [8, 9, 10]),
```

Comment:
So even though these distributions are actually different this still reproduces the case? This means its a problem that simply requires matching space datatypes only.

Reply:
Yes, but it depends on the datatypes and hyperparameters' functions. 
For example, 
1. If the function is a HP.Choice, the bug only depends on the number of the choice, not depends on what the choices are. If it has the same number of choices, it will have the bug, but if the number of choices is different, there is no bug.
2. If the function is HP.QuantUniform(label, low, high, q), it has the bug only when the distributions are identical.
3. If the function has internal function such as ":num_leaves => TreeParzen.Delayed.UnaryOperator(2 ^ TreeParzen.HP.QuantUniform(:num_leaves, 1., 4., (15 / 19)), round)," it only depends on the internal function.

Reply:
Thanks for the clarification. It seems like its related to object IDs, because these `HP.*` objects are being used as dict keys in the underlying function